var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -542,6 +543,9 @@ def setup_dependent_package(self, module, dependent_spec):
             join_path(self.prefix.lib, 'libmpi.{0}'.format(dso_suffix))
         ]
 
+    def setup_build_environment(self, env):
```

Comment:
just curious what this DISABLE_pml_rsh will do?

Reply:
I didn't find a higher level configure flag to disable support for ssh/rsh entirely, it's somewhere 200k lines deep in the configure file :grimacing:.

With +rsh:

```
$ ompi_info | grep plm
                 MCA plm: isolated (MCA v2.1.0, API v2.0.0, Component v4.1.3)
                 MCA plm: rsh (MCA v2.1.0, API v2.0.0, Component v4.1.3)
```

WIth ~rsh the last line is no longer there. I think it just skips compiling `./orte/mca/plm/rsh` altogether?

Reply:
And at runtime:

```
$ strace $(spack find --format='{prefix}' openmpi +rsh)/bin/mpirun --np 2 ./a.out |& grep ssh
...
stat("/usr/bin/ssh", {st_mode=S_IFREG|0755, st_size=789448, ...}) = 0
$ strace $(spack find --format='{prefix}' openmpi ~rsh)/bin/mpirun --np 2 ./a.out |& grep ssh
$ # nothing
```

Reply:
oh so this setup_build_environment is suppose to set env variables when the application is being run or just for building?  The configure option for not building the rsh component would be
```
--enable-mca-no-build=plm-rsh
```

Reply:
Thanks, that's better